quil-rs/src/program/graph.rs

```diff
@@ -656,8 +653,8 @@ impl ScheduledProgram {
                     &mut working_label,
                     program,
                     instruction_index,
-                ),
-            }?;
+                )?,
+            };
```

Comment:
This and above changes clarify the logic of the `match` statement, IMO, by returning errors when they happen instead of at the very end of a long statement.